src/CommunicationExecutor.c

```diff
@@ -353,7 +420,20 @@ void Ros_Communication_StartExecutors(SEM_ID semCommunicationExecutorStatus)
     //===========================================================
     //===========================================================
     //===========================================================
-    
+
+    // See if we should enable the timer which monitors user lan port link state
+    if (!g_nodeConfigSettings.userlan_monitor_enabled)
+    {
+        rc = rcl_timer_cancel(&timerMonitorUserLanState); RCL_UNUSED(rc);
```

Comment:
I don't undersatnd this cancel operation. The timer hasn't been started yet, has it?

Reply:
AFAIK, timers don't need to be started explicitly. They start 'running' as soon as the executor starts spinning.

For the `executor_motion_control` this timer is added to, that would be as soon as it enters the `while (g_Ros_Communication_AgentIsConnected)` that follows. 

We unconditionally add the timer to the executor, so if the lan state monitor was disabled via the config file, we need to disable it before the executor spins.